compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -322,7 +322,7 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         }
 
         self.suggest_bare_struct_literal(&mut err);
-        self.suggest_pattern_match_with_let(&mut err, source, span);
+        let pattern_match = self.suggest_pattern_match_with_let(&mut err, source, span);
```

Comment:
Can we simply return `(err, Vec::new())` if that call returns true?

Reply:
That is going to omit the fallback label

```diff
diff of stderr:

2         --> $DIR/if-let-typo.rs:4:13
3          |
4       LL |     if Some(x) = foo {}
-          |             ^ not found in this scope
+          |             ^
6          |
7       help: you might have meant to use pattern matching
8          |

13        --> $DIR/if-let-typo.rs:10:8
14         |
15      LL |     if x = 5 {}
-          |        ^ not found in this scope
+          |        ^
17         |
18      help: you might have meant to use pattern matching
19         |
```